### PR TITLE
feat: pin node version

### PR DESCRIPTION
**Pull request by doug-s-nava**
pins node version at 14.20.0 for local and ci builds, and adjusts readmes to include information on usage of nvm

See:
* https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#node-version-file

## Test Steps:
1. look at latest frontend build for this branch
2. _VERIFY_: build ran on version 14.20 and passed as expected
3. also ensure that building and testing on your local machine works as expected, following directions from readmes as necessary to use nvm to install version 14.20

Screen Shot 2022-09-29 at 9 57 02 AM

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

### Testing
- [ ] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

**Comment by snesm** [comment]
Node 14 is approaching end of life: https://github.com/CDCgov/prime-reportstream/issues/5926

File: frontend-react/README.md [hunk 1]
```diff
@@ -1,15 +1,19 @@
 # Getting started with ReportStream's React application
+
 ## Run the React application
 
 Our new React front-end is easy to get up and running on your machine. First, ensure the following dependencies
 installed:
-- `node` (version 14.x)
-- `yarn` package manager
+
```
Comment (doug-s-nava) [inline frontend-react/README.md:7]: most of this is auto format corrections, but this part is new

File: prime-router/docs/getting-started/Using-an-apple-silicon-mac.md [hunk 2]
```diff
@@ -105,13 +108,16 @@ You should be able to run tests now to confirm that everything is working.
 
 Our new React front-end is easy to get up and running on your machine. First, ensure the following dependencies
 installed:
-- `node` (version 14.x)
-- `yarn` package manager
+
+-   `node` (version 14.20.0) via `nvm`
```
Comment (doug-s-nava) [inline prime-router/docs/getting-started/Using-an-apple-silicon-mac.md:112]: again, most of this is formatting auto fixes, this is the salient piece

**Review by snesm — CHANGES_REQUESTED** [review]
Why do we want to pin the minor version?  

If there is a 14.21 it is likely to be a security fix that we should upgrade to.

**Comment by doug-s-nava** [comment]
we have had some npm packages cause errors due to mismatches in minor versions, so good to keep that tight. I am open to other thoughts if the maintenance side of that is deemed too intense though

**Review by snesm — CHANGES_REQUESTED** [review]
14.20.1 was released in Sep. and contained security fixes.  Please update.

File: frontend-react/package.json [hunk 3]
```diff
@@ -228,5 +228,8 @@
     "resolutions": {
         "@types/react": "<18.0.0",
         "@types/react-dom": "<18.0.0"
+    },
+    "engines": {
+      "node": "~14.20.0"
```
Comment (snesm) [inline frontend-react/package.json:233]: Possible conflict with .1 used elsewhere?

**Review by snesm — CHANGES_REQUESTED** [review]

File: frontend-react/package.json (hunk 3 above)
Comment (snesm) [reply frontend-react/package.json:233]: @doug-s-nava this allows "approximate" version, which will get 14.20.1 for now, but may get newer versions in the future.  Will that conflict with the hard coded 14.20.1 used elsewhere?

Comment (doug-s-nava) [reply frontend-react/package.json:233]: in my mind it is fine / preferable to be more permissive here. The real source of truth for node version in use will be the nvmrc, any github action specs, and hopefully down the line, dockerfiles. This engines definition will cause npm installs to error if a version outside of the one specified is used, but if all of the above mentioned files are pinned more specifically, as they should be, that will be fine. This just gives local developers a little bit more leeway, not forcing them to install new patch versions all the time to keep up.

On a procedural note, I'd love to have this wrapped up this morning, as Monday is my last day and I'd like to get the team onboarded to nvm before I leave

**Review by snesm — APPROVED** [review]

**Pull request merged**